### PR TITLE
chore(DataStore): integration test code fixes

### PR DESCRIPTION
**Pull request by lawmicha**
*Issue #, if available:*

*Description of changes:*
Small changes that fixes some of the integration tests fixed up under AWSDataStoreIntegrationTests

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/DataStoreEndToEndTests.swift [hunk 1]
```diff
@@ -150,13 +146,61 @@ class DataStoreEndToEndTests: SyncEngineIntegrationTestBase {
...
 
                 if mutationEvent.mutationType == GraphQLMutationType.update.rawValue {
                     XCTAssertEqual(post.content, updatedPost.content)
                     XCTAssertEqual(mutationEvent.version, 2)
                     updateReceived.fulfill()
                     return
                 }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
```
Comment (ameter) [inline AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/DataStoreEndToEndTests.swift:182]: Recommend we update `HubListenerTestUtilities.waitForListener(...)` to be async and then use `try await Task.sleep(seconds:)` instead of `Thread.sleep(forTimeInterval:)` in that function.  This will make it create a Swift Concurrency suspension instead of blocking the Thread.  That can be a separate PR though.

Comment (ameter) [reply AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/DataStoreEndToEndTests.swift:182]: see [this PR](https://github.com/aws-amplify/amplify-ios/pull/2084) for a convenience wrapper

**Review by ameter — APPROVED** [review]
Approved with comment.

File: AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/DataStoreEndToEndTests.swift (hunk 1 above)
Comment (lawmicha) [reply AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/DataStoreEndToEndTests.swift:182]: Thanks! I checked the usage of this and it spans a couple of places, so will cut a new PR for this

Comment (lawmicha) [reply AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/DataStoreEndToEndTests.swift:182]: https://github.com/aws-amplify/amplify-ios/pull/2087
